src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlParameter.cs

```diff
@@ -978,9 +1005,15 @@ private void CloneHelper(SqlParameter destination)
             destination._offset = _offset;
             destination._sourceColumn = _sourceColumn;
             destination._sourceVersion = _sourceVersion;
-            destination._sourceColumnNullMapping = _sourceColumnNullMapping;
-            destination._isNullable = _isNullable;
-
+            destination._flags = _flags & (
```

Comment:
This mask is created from the field which were being individually set in the clone method. I think it might need some things adding, like IsDerivedParameterTypeName and possibly HasScale.